File: ResourceManager/Helpers/AssetManager.cs
using System;
using System.Windows.Media.Imaging;

namespace ResourceManager.Helpers
{
    internal class AssetManager
    {
        private static readonly AssetManager _instance = new AssetManager();
        private AssetManager()
        { }
        public static AssetManager Instance => _instance;
        public BitmapImage GetImage(string resourceName)
        {
            var image = new BitmapImage();

            string resourceLocation =
                string.Format("pack://application:,,,/ResourceManager;component/Resources/Assets/Images/{0}", resourceName);

            try
            {
                image.BeginInit();
                image.CacheOption = BitmapCacheOption.OnLoad;
                image.CreateOptions = BitmapCreateOptions.IgnoreImageCache;
                image.UriSource = new Uri(resourceLocation);
                image.EndInit();
            }
            catch (Exception e)
            {
                System.Diagnostics.Trace.WriteLine(e.ToString());
            }

            return image;
        }
    }
}

File: ResourceManager/Helpers/AssetManager.vb
Imports System
Imports System.Windows.Media.Imaging

Namespace ResourceManager.Helpers
    Friend Class AssetManager
        Private Shared ReadOnly _instance As AssetManager = New AssetManager()
        Private Sub New()
        End Sub
        Public Shared ReadOnly Property Instance As AssetManager
            Get
                Return _instance
            End Get
        End Property
        Public Function GetImage(ByVal resourceName As String) As BitmapImage
            Dim image = New BitmapImage()

            Dim resourceLocation = String.Format("pack://application:,,,/ResourceManager;component/Resources/Assets/Images/{0}", resourceName)

            Try
                image.BeginInit()
                image.CacheOption = BitmapCacheOption.OnLoad
                image.CreateOptions = BitmapCreateOptions.IgnoreImageCache
                image.UriSource = New Uri(resourceLocation)
                image.EndInit()
            Catch e As Exception
                Trace.WriteLine(e.ToString())
            End Try

            Return image
        End Function
    End Class
End Namespace
